include/simdjson/dom/array-inl.h

```diff
@@ -53,18 +53,38 @@ namespace dom {
 simdjson_inline array::array() noexcept : tape{} {}
 simdjson_inline array::array(const internal::tape_ref &_tape) noexcept : tape{_tape} {}
 inline array::iterator array::begin() const noexcept {
+#if SIMDJSON_DEVELOPMENT_CHECKS
+  // issue https://github.com/simdjson/simdjson/issues/1914
+  assert (tape.usable()); // the default constructed array is invalid
+#endif
```

Comment:
This could be made shorter and avoid the ifdefs with a macro for SIMDJSON_DEVELOPMENT_ASSERT or SIMDJSON_ASSERT_TAPE_USABLE - in the latter case, the code comments could go on the macro

Still getting familiar with the coding conventions.



```c
#if SIMDJSON_DEVELOPMENT_CHECKS
#define SIMDJSON_DEVELOPMENT_ASSERT(expr) do { assert ((expr)); } while (0)
#else
#define SIMDJSON_DEVELOPMENT_ASSERT(expr) do { } while (0)
#endif
```

Reply:
Agreed. I stole your macro.